charts/prometheus-modbus-exporter/values.yaml

```diff
@@ -0,0 +1,165 @@
+# Default values for prometheus-modbus-exporter.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+## if one needs to define the below modbus.yml in a separate configMap
+## that configMap name can be mentioned here:
+# configMapFile: "my-existing-modbus-config"
+
+## when the above configMapFile parameter is not defined, it will be created
+## using below parameters (usually send to the chart as custom values)
+modules:
+# Module name, needs to be passed as parameter by Prometheus.
+# the name must be matched with the name of serviceMonitor.endpointsConfig[x].params.module
+- name: "fake"
+  protocol: 'tcp/ip'
+  metrics:
+  # Name of the metric.
+  - name: "power_consumption_total"
+    # Help text of the metric.
+    help: "represents the overall power consumption by phase"
+    # Labels to be added to the time series.
+    labels:
+      phase: "1"
+    # Register address.
+    # The first digit of the address is the function code
+    # Supported codes are: 1, 2, 3, 4
+    address: 300022
+    # Datatypes allowed: bool, int16, int32, int64, uint16, uint32, uint64,
+    #   float16, float32, float64
+    # One register holds 16 bits.
+    dataType: int16
+    # Endianness allowed: big, little, mixed, yolo
+    # Optional. If not defined: big.
+    endianness: big
+    # Prometheus metric type: https://prometheus.io/docs/concepts/metric_types/.
+    metricType: counter
+    # Factor can be specified to represent metric value.
+    # Examples: 1, 2, 1.543, 0.01 etc
+    # Factor is multiplied with the scraped value to produce the metric value
+    # Optional.
+    factor: 3.1415926535
+
+  - name: "some_gauge"
+    help: "some help for some gauge"
+    address: 30023
+    dataType: int16
+    metricType: gauge
+    factor: 2
+
+  - name: "coil"
+    help: "some help for some coil"
+    address: 124
+    dataType: bool
+    bitOffset: 0
+    metricType: gauge
+
+## The prometheus operator serviceMonitor for modbus
+serviceMonitor:
+  ## even if enabled is true, if serviceMonitor crd is not installed, this will be skipped.
+  enabled: true
+  endpointsConfig:
+  - params:
+      target:
+      - "10.1.2.3:502"
```

Comment:
setting some ip her seems not to make sense as this will likely not work as a default setting for most installations?

Reply:
This is a value to be populated, but to keep these in sync with the default config. found in https://github.com/RichiH/modbus_exporter/, I've kept the same.
Note: some value is required, otherwise helm tests will fail.

Reply:
But the ip is very likely non existend.
It makes no sense to configure it.